rc/filetype/fsharp.kak

```diff
@@ -0,0 +1,126 @@
+# https://fsharp.org/
+#
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*[.](fs|fsx|fsi) %{
+    set-option buffer filetype fsharp
+}
+
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+hook global WinSetOption filetype=fsharp %{
+    require-module fsharp
+
+    # cleanup trailing whitespaces upon exiting insert mode
+    hook window ModeChange insert:.* -group fsharp-trim-indent %{ try %{ execute-keys -draft \; <a-x> s ^\h+$ <ret> d } }
```

Comment:
We tend to avoid this hook (which might still be lying around in some form or another in the codebase). Consider the following usecase:

* yank some text
* start writing `let x = ` (trailing whitespace)
* leave insert mode, which triggers the hook
* paste

The outcome would be `let x=[text]`, when what we wanted was really `let x = [text]` (whitespace around the operator).

I think a better way of handling whitespace errors is to use a formatter (c.f. `:format`, `formatcmd`); or you could set a highlighter in your configuration that shows trailing whitespace:

```
add-highlighter global/ regex '\h+$' 0:Error
```

Reply:
Right, I have wondered about this behaviour. Most filetypes seem to have it -- I have grown used to pasting _after_ (`p`), s.t. the paste lands on a new line, and then concatenating the lines with `<s-J>`.

I will remove this line from the file and keep the formatter option in mind.